### PR TITLE
Dr.B's Requested Changes

### PR DESCRIPTION
**Pull request by Sech1**
Please review these changes and let me know if anything else needs updated/modified.
These are the changes Dr.B wanted me to perform, and the additional data he wanted when the form was submitted.

If anything is incorrect or you have suggestions please let me know.

### CODE REVIEW

**Comment by Sech1** [comment]
I have also resolved any merge conflicts.

**Comment by Sech1** [comment]
Made some updates to fix crash if guardian values were left blank. 
This may change as Dr.B didn't specify if these should be allowed to be left blank or not.

File: resources/views/join.blade.php [hunk 1]
```diff
@@ -49,14 +62,60 @@
...
+
+      <!-- Guardian Email Field -->
+      <div class="field">
+        <label class="label">Parent / Guardian Email</label>
+        <p class="control has-icons-left has-icons-right">
+          <input id="guardianEmail" class="input" type="email" placeholder="Parent or Guardian Email" name="guardianEmail" onchange="hideErrors()">
+          <span class="icon is-small is-left">
+        <i class="fa fa-envelope"></i>
+      </span>
+          <span class="icon is-small is-right">
```
Comment (ryanvade) [inline resources/views/join.blade.php]: I'm thinking we remove the `<span class="icon is-small is-right">
        <i class="fa fa-check"></i>
      </span>`

From both email fields.

File: resources/views/join.blade.php [hunk 2]
```diff
@@ -49,14 +62,60 @@
...
+        <i class="fa fa-check"></i>
+      </span>
+        </p>
+      </div>
+      <!-- END Guardian Email Field -->
+
+      <!-- Guardian Phone Number -->
+      <div class="field">
+        <label class="label">Parent / Guardian Phone Number</label>
+        <div class="control">
```
Comment (ryanvade) [inline resources/views/join.blade.php]:
```suggestion
        <div class="control has-icons-left">
```

And then after the phone input:
`<span class="icon is-small is-lefft">
 	<i class="fa fa-phone" aria-hidden="true"></i>
 </span>`

File: resources/views/join.blade.php (hunk 1 above)
Comment (ryanvade) [inline resources/views/join.blade.php]: Lets also change 
`<div class="column is-half-desktop has-text-centered">`
on line 8 to
`<div class="column is-two-thirds-desktop">`

As the increased amount of text and download links don't look terribly good centered with only half the screen.

File: resources/views/join.blade.php [hunk 3]
```diff
@@ -31,12 +31,25 @@
 @endif
     <form action="/join" class="" method="POST" enctype="multipart/form-data" id="form">
       {{ csrf_field() }}
+      <!-- First Name Field -->
       <div class="field">
         <label class="label">Name</label>
         <div class="control">
-          <input id="name" class="input" type="text" placeholder="First Last" name="name" onchange="hideErrors()">
+          <input id="name" class="input" type="text" placeholder="First Name" name="name" onchange="hideErrors()">
```
Comment (ryanvade) [inline resources/views/join.blade.php]:
```suggestion
          <input id="name" class="input" type="text" placeholder="First Name" name="name" onchange="hideErrors()" required>
```
I forgot this as well on my version.

File: resources/views/join.blade.php [hunk 4]
```diff
@@ -31,12 +31,25 @@
...
         </div>
       </div>
+      <!-- END First Name Field -->
+
+      <!-- Last Name Field -->
+      <div class="field">
+        <label class="label">Last</label>
+        <div class="control">
+          <input id="last" class="input" type="text" placeholder="Last Name" name="last" onchange="hideErrors()">
```
Comment (ryanvade) [inline resources/views/join.blade.php]:
```suggestion
          <input id="last" class="input" type="text" placeholder="Last Name" name="last" onchange="hideErrors()" required>
```

Comment (ryanvade) [inline resources/views/join.blade.php]: Don't forget to add required to the student email input as well.

File: app/Http/Controllers/SignupController.php [hunk 5]
```diff
@@ -42,29 +42,64 @@ public function store(Request $request)
...
           'name' => 'required',
+          'last' => 'required',
+          'guardianFirst' => 'nullable',
+          'guardianLast' => 'nullable',
+          'guardianEmail' => 'nullable|email',
+          'guardianNumber' => 'nullable|regex:/^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}$/',
           'email' => 'required|email',
-	        'shirts' => 'required',
+            'shirts' => 'required',
           'allergies' => 'required',
```
Comment (ryanvade) [inline app/Http/Controllers/SignupController.php:52]: So right now, it is impossible to submit the form without *something* in the text area. We should either not make it required setup some kind of notice on the client side. OR add a checkbox for 'My student doesn't have allergies'..something. So we can do a [required_unless](https://laravel.com/docs/5.5/validation#rule-required-unless) on the textarea with the checkbox.

File: resources/views/join.blade.php (hunk 4 above)
Comment (ryanvade) [reply resources/views/join.blade.php]: And the shirts technically...my bad on these.

File: app/Http/Controllers/SignupController.php [hunk 6]
```diff
@@ -42,29 +42,64 @@ public function store(Request $request)
...
+            $path = $request->file('doc')->storeAs(
           'documents',
                   str_slug($request->name, '') . '.' . $request->file('doc')->getClientOriginalExtension()
           );
         } else {
-          $path = 'no file';
+            $path = 'no file';
+        }
+
+        if (!empty($request->input('guardianFirst'))) {
```
Comment (ryanvade) [inline app/Http/Controllers/SignupController.php]: Lots of if/else statements.  A bit hard to read don't you think?

We can replace with:
`$guardianFirstVar = $request->input('guardianFirst') ?? 'No Parent or Guardian Entered';`

Uses the isset ternary/null coalesce operation available with PHP 7.

File: app/Http/Controllers/SignupController.php [hunk 7]
```diff
@@ -42,29 +42,64 @@ public function store(Request $request)
...
+            $guardianFirstVar = $request->input('guardianFirst');
+        } else {
+            $guardianFirstVar = 'No Parent or Guardian Entered';
+        }
+
+        if (!empty($request->input('guardianLast'))) {
+            $guardianLastVar = $request->input('guardianLast');
```
Comment (ryanvade) [inline app/Http/Controllers/SignupController.php]: ``$guardianLastVar = $request->input('guardianLast') ?? 'No Parent or Guardian Entered';``

File: app/Http/Controllers/SignupController.php [hunk 8]
```diff
@@ -42,29 +42,64 @@ public function store(Request $request)
...
+        } else {
+            $guardianLastVar = 'No Parent or Guardian Entered';
         }
 
+        if (!empty($request->input('guardianEmail'))) {
+            $guardianEmailVar = $request->input('guardianEmail');
```
Comment (ryanvade) [inline app/Http/Controllers/SignupController.php]: `$guardianEmailVar = $request->input('guardianEmail') ?? 'No Parent or Guardian Email Entered';`

File: app/Http/Controllers/SignupController.php [hunk 9]
```diff
@@ -42,29 +42,64 @@ public function store(Request $request)
...
+        } else {
+            $guardianEmailVar = 'No Parent or Guardian Email Entered';
+        }
+
+        if (!empty($request->input('guardianNumber'))) {
+            $guardianNumberVar = $request->input('guardianNumber');
```
Comment (ryanvade) [inline app/Http/Controllers/SignupController.php]: `$guardianNumberVar = $request->input('guardianNumber') ?? 'No Parent or Guardian Number Entered';`

File: app/Http/Controllers/SignupController.php (hunk 5 above)
Comment (ryanvade) [reply app/Http/Controllers/SignupController.php:52]: Technically, we should probably be validating the shirts with [required_in](https://laravel.com/docs/5.5/validation#rule-in)

But it is probably fine.

**Review by ryanvade — CHANGES_REQUESTED** [review]
The allergies input needs to be fixed.  Requiring a textarea is bad.

File: resources/views/join.blade.php (hunk 1 above)
Comment (ryanvade) [reply resources/views/join.blade.php]: And also the `has-icons-right` class from those inputs.

**Comment by ryanvade** [comment]
Apologies, for the late response. I can implement and resolve if you want (just say the word).  BUT I will need input on the allergies input.

**Comment by Sech1** [comment]
The requested changes have been made, I'll have to talk to Dr.B about how he wants to handle the allergies and students over the age of 18.
We may include a couple check boxes that hide the forms that aren't necessary if someone is over the age of 18 (Doesn't need a parent/guardian) or has no allergies. 

Let me know what you think about this going forward, I hopefully will be able to meet with Dr.B sometime this week.

**Comment by ryanvade** [comment]
Those other changes need to be discussed before putting it live.

**Pull request merged**